cpp/src/parquet/arrow/schema.cc

```diff
@@ -427,6 +428,13 @@ Status FieldToNode(const std::string& name, const std::shared_ptr<Field>& field,
     }
     case ArrowTypeId::EXTENSION: {
       auto ext_type = std::static_pointer_cast<::arrow::ExtensionType>(field->type());
+      // Built-in JSON extension is handled differently.
+      if (ext_type->extension_name() == std::string("arrow.json")) {
```

Comment:
So the writer-side, this is forced to:

```
      type = ParquetType::BYTE_ARRAY;
      logical_type = LogicalType::JSON();
```

On reader side, it's set by `get_arrow_extensions_enabled()`

Reply:
Yes. Are you just noting or proposing a change? :)